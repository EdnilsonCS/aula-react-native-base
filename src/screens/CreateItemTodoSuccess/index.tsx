import { useNavigation } from '@react-navigation/native';
import React, { useCallback } from 'react';
import { Button } from '@components/index';
import { Container, Title, Description } from './styles';

const CreateItemTodoSuccess: React.FC = () => {
  const navigation = useNavigation();
  const handleToNavigationToNewItem = useCallback(() => {
    navigation.navigate('ListTodo');
  }, [navigation]);
  return (
    <Container>
      <Title>Parab√©ns</Title>
      <Description>Seu item foi criado com sucesso!</Description>
      <Button onPress={handleToNavigationToNewItem}>Ok!</Button>
    </Container>
  );
};

export default CreateItemTodoSuccess;
