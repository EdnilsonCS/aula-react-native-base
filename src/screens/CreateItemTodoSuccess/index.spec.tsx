import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import CreateItemTodoSuccess from './index';

const mockedHistoryPush = jest.fn();

jest.mock('@react-navigation/native', () => {
  return {
    useNavigation: () => ({
      goBack: jest.fn(),
      navigate: mockedHistoryPush,
    }),
  };
});

describe('CreateItemTodoSuccess', () => {
  it('Should be able to render correct', () => {
    const tree = render(<CreateItemTodoSuccess />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should be able to navigate to todo list ', () => {
    const { getByText } = render(<CreateItemTodoSuccess />);
    const buttonElement = getByText('Ok!');

    fireEvent.press(buttonElement);

    expect(mockedHistoryPush).toBeCalledWith('ListTodo');
  });
});
