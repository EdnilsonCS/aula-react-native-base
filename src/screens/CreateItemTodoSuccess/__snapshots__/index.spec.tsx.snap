// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateItemTodoSuccess Should be able to render correct 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 14,
        },
      ]
    }
  >
    Parab√©ns
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontSize": 14,
        },
      ]
    }
  >
    Seu item foi criado com sucesso!
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#1F1C1C",
        "borderRadius": 5,
        "justifyContent": "center",
        "minWidth": 96,
        "opacity": 1,
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 14,
            "letterSpacing": 1.25,
            "lineHeight": 16,
          },
        ]
      }
    >
      Ok!
    </Text>
  </View>
</View>
`;
